mlibrary/lib.umich.edu
# [WEBSITE-67/69/72/73/76] Package Updates

To improve the maintenance and security of [lib.umich.edu](https://www.lib.umich.edu/).

## `gatsby-codemods`
`gatsby-codemods` has been removed as it was used to help with major version Gatsby migrations, and is no longer used.

## `react-helmet`
`gatsby-plugin-react-helmet` [has been deprecated](https://www.gatsbyjs.com/plugins/gatsby-plugin-react-helmet/) and is now natively supported by the [Gatsby Head API](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-head/). The plugin also relied on `react-helmet`, so both of those dependencies have been removed. In order to apply the `lang` attribute for the `html` tag, `gatsby-ssr.js` needed to be created. To see the file in development, the `DEV_SSR` flag was added to `gatsby-config.js`.

## `gatsby-alias-imports`
`gatsby-alias-imports` was only used to create one alias: `@reusable`. All files that used the alias have been replace with the actual path, and the dependency has been removed.

## `@reach`
The `@reach` UI still lacks support for React v18. The latest version of `gatsby` enforces the latest version of `react`, so all dependencies have been uninstalled/replaced.

### `@reach/alert`
`@reach/alert` has been replaced with accessible and styled `div` elements to reduce dependencies.

### `@reach/combobox`
`@reach/combobox` has been removed from not being used.

### `@reach/dialog`
`@reach/dialog` has been replaced with the native HTML `dialog` element. Styles and event listeners have also been added to mimic the original component.

### `@reach/window-size`
`@reach/window-size` has been replaced with a custom `WindowResize` function.

## Testing
- Remove the packages (`rm -rf node_modules && rm package-lock.json`) , install the dependencies (`npm install`), and start the site (`npm start`).
  - Poke around [localhost:8000](http://localhost:8000/) and see if anything is broken.
- Make sure the PR is consistent in these browsers:
